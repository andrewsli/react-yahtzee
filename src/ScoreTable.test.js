import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import ScoreTable from "./ScoreTable";

const scoreTableProps = {
  scores: {
    ones: 2,
    twos: 4,
    threes: 6,
    fours: 8,
    fives: 10,
    sixes: 12,
    threeOfKind: 3,
    fourOfKind: 4,
    fullHouse: 25,
    smallStraight: 30,
    largeStraight: 40,
    yahtzee: 50,
    chance: 5
  }
}

it("renders ScoreTable without crashing", function () {
  shallow(<ScoreTable scores={scoreTableProps.scores} />);
});

it("matches snapshot", function () {
  let wrapper = shallow(<ScoreTable scores={scoreTableProps.scores} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

