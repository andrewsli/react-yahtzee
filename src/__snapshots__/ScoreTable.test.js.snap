// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="ScoreTable"
>
  <section
    className="ScoreTable-section"
  >
    <h2>
      Upper
    </h2>
    <table
      cellSpacing="0"
    >
      <tbody>
        <RuleRow
          doScore={[Function]}
          name="Ones"
          score={2}
        />
        <RuleRow
          doScore={[Function]}
          name="Twos"
          score={4}
        />
        <RuleRow
          doScore={[Function]}
          name="Threes"
          score={6}
        />
        <RuleRow
          doScore={[Function]}
          name="Fours"
          score={8}
        />
        <RuleRow
          doScore={[Function]}
          name="Fives"
          score={10}
        />
        <RuleRow
          doScore={[Function]}
          name="Sixes"
          score={12}
        />
      </tbody>
    </table>
  </section>
  <section
    className="ScoreTable-section ScoreTable-section-lower"
  >
    <h2>
      Lower
    </h2>
    <table
      cellSpacing="0"
    >
      <tbody>
        <RuleRow
          doScore={[Function]}
          name="Three of Kind"
          score={3}
        />
        <RuleRow
          doScore={[Function]}
          name="Four of Kind"
          score={4}
        />
        <RuleRow
          doScore={[Function]}
          name="Full House"
          score={25}
        />
        <RuleRow
          doScore={[Function]}
          name="Small Straight"
          score={30}
        />
        <RuleRow
          doScore={[Function]}
          name="Large Straight"
          score={40}
        />
        <RuleRow
          doScore={[Function]}
          name="Yahtzee"
          score={50}
        />
        <RuleRow
          doScore={[Function]}
          name="Chance"
          score={5}
        />
      </tbody>
    </table>
  </section>
</div>
`;
